ouroboros-consensus/src/Ouroboros/Consensus/Mempool/Impl/Pure.hs

```diff
@@ -0,0 +1,321 @@
+-- point changes.
+runSyncWithLedger :: forall m blk. IOLike m
+  => StrictTVar m (InternalState blk)
+  -> SyncWithLedger blk
+  -> STM m (Maybe (TraceEventMempool blk), MempoolSnapshot blk TicketNo)
+runSyncWithLedger stateVar (NewSyncedState is msp mTrace) = do
+  writeTVar stateVar is
+  return (mTrace, msp)
+
+-- | Create a 'SyncWithLedger' value representing the values that will need to
+-- be stored for committing this synchronization with the Ledger.
+--
+-- See the documentation of 'runSyncWithLedger' for more context.
+pureSyncWithLedger :: (LedgerSupportsMempool blk, HasTxId (GenTx blk), ValidateEnvelope blk)
+                   => InternalState blk
+                   -> LedgerState blk
+                   -> LedgerConfig blk
+                   -> MempoolCapacityBytesOverride
+                   -> SyncWithLedger blk
+pureSyncWithLedger istate lstate lcfg capacityOverride =
+  let vr = validateIS istate lstate lcfg capacityOverride
+      mempoolSize = isMempoolSize istate
+      snapshot = implSnapshotFromIS istate
+      removed = map fst (vrInvalid vr)
+      mTrace = if null removed
+               then Nothing
+               else Just $
+                    TraceMempoolRemoveTxs removed mempoolSize
+  in
+    NewSyncedState (internalStateFromVR vr) snapshot mTrace
+
+-- | Get a snapshot of the mempool state that is valid with respect to
+-- the given ledger state
+pureGetSnapshotFor :: forall blk.
+                      ( LedgerSupportsMempool blk
+                      , HasTxId (GenTx blk)
+                      , ValidateEnvelope blk
+                      )
+                   => LedgerConfig blk
+                   -> ForgeLedgerState blk
+                   -> MempoolCapacityBytesOverride
+                   -> InternalState blk
+                   -> MempoolSnapshot blk TicketNo
+pureGetSnapshotFor cfg blockLedgerState capacityOverride =
+    implSnapshotFromIS
+    . internalStateFromVR
+    . validateStateFor capacityOverride cfg blockLedgerState
+
+{-------------------------------------------------------------------------------
+  MempoolSnapshot Implementation
+-------------------------------------------------------------------------------}
+
+-- | Create a Mempool Snapshot from a given Internal State of the mempool.
+implSnapshotFromIS :: HasTxId (GenTx blk)
+                   => InternalState blk
+                   -> MempoolSnapshot blk TicketNo
+implSnapshotFromIS is = MempoolSnapshot {
+      snapshotTxs         = implSnapshotGetTxs         is
+    , snapshotTxsAfter    = implSnapshotGetTxsAfter    is
+    , snapshotTxsForSize  = implSnapshotGetTxsForSize  is
+    , snapshotLookupTx    = implSnapshotGetTx          is
+    , snapshotHasTx       = implSnapshotHasTx          is
+    , snapshotMempoolSize = implSnapshotGetMempoolSize is
+    , snapshotSlotNo      = isSlotNo                   is
+    , snapshotLedgerState = isLedgerState              is
+    }
+ where
+  implSnapshotGetTxs :: InternalState blk
```

Comment:
Style: the `::` and `->` in these where clause are off.

Reply:
Indeed (I <TAB>ed the beginning of the type signature and forgot to move the rest). Fixed (resolving)